src/swarm-tui/Swarm/TUI/View/Robot.hs

```diff
@@ -28,17 +39,159 @@ import Swarm.Game.State.Substate
 import Swarm.Game.Tick (addTicks)
 import Swarm.Game.Universe
 import Swarm.Game.World.Coords
-import Swarm.TUI.Model
-import Swarm.TUI.Model.DebugOption (DebugOption (..))
-import Swarm.TUI.Model.UI
+import Swarm.TUI.Model.DebugOption
+import Swarm.TUI.Model.Name
+import Swarm.TUI.Model.UI.Gameplay
 import Swarm.TUI.View.Attribute.Attr
 import Swarm.TUI.View.CellDisplay
-import Swarm.TUI.View.Util as VU
-import Swarm.Util
+import Swarm.TUI.View.RobotDisplay
+import Swarm.Util (applyWhen, maximum0)
 import Swarm.Util.UnitInterval
 import Swarm.Util.WindowedCounter qualified as WC
 import System.Clock (TimeSpec (..))
 
+instance Semigroup ColWidth where
+  ColW w1 <> ColW w2 = ColW $ max w1 w2
+
+data RobotRenderingContext = RobotRenderingContext
+  { _mygs :: GameState
+  , _gameplay :: UIGameplay
+  , _timing :: UITiming
+  , _uiDbg :: Set DebugOption
+  }
+
+makeLenses ''RobotRenderingContext
+
+mkRobotDisplay :: RobotRenderingContext -> RobotListContent
+mkRobotDisplay c =
+  RobotListContent
+    { _libList = mixedTabularList RobotsList (mkLibraryEntries c) (LstItmH 1) (wprk uiDebug) wpr
+    , _libRenderers =
```

Comment:
Don't re-initialize the `_libRenderers` field, since it's constant with respect to all rendering context other than `uiDbg`